### PR TITLE
Roles allowed test

### LINKED ISSUE #12050: @RolesAllowed rejects unauthenticated users when they mapped to an allowed (EVERYONE) role
**Describe the bug**
The special-subject `EVERYONE` doesn't play well with the `@RolesAllowed` annotation on JAX-RS resources.

**Steps to Reproduce**
1. Define a simple webapp with a single JAX-RS resource such as this one thats based on the guide:
  ```
package io.openliberty.guides.rest;

import java.util.Properties;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("properties")
public class PropertiesResource {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Properties getProperties() {
        return System.getProperties();
    }

}
  ```

2. Add a `@RolesAllowed("User")` annotation to this class and bind the special-subject `EVERYONE` to this security-role:
  ```
<server description="Auth issue example">

  <featureManager>
      <feature>jaxrs-2.1</feature>
      <feature>appSecurity-2.0</feature>
  </featureManager>

  <httpEndpoint httpPort="${default.http.port}" httpsPort="${default.https.port}" host="*"/>

  <webApplication location="guide-rest-intro.war" contextRoot="${app.context.root}">
    <application-bnd>
      <security-role name="User">
        <special-subject type="EVERYONE"/>
      </security-role>
    </application-bnd>
  </webApplication>

  <webAppSecurity singleSignonEnabled="false"/>
</server>
  ```

3. Invoke the endpoint (e.g. http://localhost:9080/LibertyProject/System/properties) without passing auth information.  Note that the server rejects the request with status of HTTP 401.

**Expected behavior**
When the special-subject EVERYONE is bound to a security-role, unauthenticated users should be considered to be in that role and should have access to the endpoints allowed for that role.

**Diagnostic information:**
 - OpenLiberty Version: 20.0.0.4
 - Java Version:
  ```
openjdk version "11.0.5" 2019-10-15
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.17.0, JRE 11 Mac OS X amd64-64-Bit Compressed References 20191016_371 (JIT enabled, AOT enabled)
OpenJ9   - 77c1cf708
OMR      - 20db4fbc
JCL      - 2a7af5674b based on jdk-11.0.5+10)
  ```

**Additional context**
If I remove the `@RolesAllowed("User")` annotation, I am now able to access this endpoint, even when my web.xml includes a security-constraint like the following:
```
    <security-constraint>
        <web-resource-collection>
            <web-resource-name>Properties</web-resource-name>
            <url-pattern>/*</url-pattern>
            <http-method>GET</http-method>
        </web-resource-collection>
        <auth-constraint>
            <role-name>User</role-name>
        </auth-constraint>
    </security-constraint>
```
If I remove the special-subject mapping and keep this in the web.xml, it returns a 401 as expected.  So it seems that special-subject works as expected with the `auth-constraint` in web.xml.

Similarly, if I add a basicRegistry to the server.xml, then this user is able to access the resource, despite not being explicitly mapped to the seurity-role.
```
  <basicRegistry> 
    <user name="a" password="a"/>
  </basicRegistry>
```
So it seems like special-subject EVERYONE is actually behaving the way I would expect special-subject ALL_AUTHENTICATED_USERS to behave.

### PR DESCRIPTION
**Pull request by WhiteCat22**
`@RolesAllowed` on a JAX-RS endpoint does not authenticate when role has `special-subject` EVERYONE

fixes https://github.com/OpenLiberty/open-liberty/issues/12050

### CODE REVIEW

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#run-libby-bot
#build

File: dev/com.ibm.ws.jaxrs.2.1.common/bnd.bnd [hunk 1]
```diff
@@ -86,41 +86,44 @@ Include-Resource:\
 instrument.classesExcludes: com/ibm/ws/jaxrs20/internal/resources/*.class
 
 -buildpath: \
-  com.ibm.websphere.javaee.jaxb.2.2;version=latest,\
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxrs.2.1.common/bnd.bnd:89]: Add 2022 to copyright

File: dev/com.ibm.ws.jaxrs.2.1.common/src/com/ibm/ws/jaxrs20/security/LibertyAuthFilter.java [hunk 2]
```diff
@@ -29,19 +29,26 @@
 import org.apache.cxf.service.invoker.MethodDispatcher;
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxrs.2.1.common/src/com/ibm/ws/jaxrs20/security/LibertyAuthFilter.java:29]: Add 2022 to copyright

File: dev/com.ibm.ws.jaxrs.2.1_fat/test-applications/RolesAllowed/src/com/ibm/ws/jaxrs/fat/rolesallowed/RolesAllowedApplication.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxrs.2.1_fat/test-applications/RolesAllowed/src/com/ibm/ws/jaxrs/fat/rolesallowed/RolesAllowedApplication.java]: Add 2022

File: dev/com.ibm.ws.jaxrs.2.1_fat/test-applications/RolesAllowed/src/com/ibm/ws/jaxrs/fat/rolesallowed/RolesAllowedResource.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxrs.2.1_fat/test-applications/RolesAllowed/src/com/ibm/ws/jaxrs/fat/rolesallowed/RolesAllowedResource.java]: Add 2022

File: dev/com.ibm.ws.jaxrs.2.1_fat/test-applications/RolesAllowed/src/com/ibm/ws/jaxrs/fat/rolesallowed/servlet/RolesAllowedTestServlet.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxrs.2.1_fat/test-applications/RolesAllowed/src/com/ibm/ws/jaxrs/fat/rolesallowed/servlet/RolesAllowedTestServlet.java]: Add 2022

File: dev/com.ibm.ws.security.authorization.util/src/com/ibm/ws/security/authorization/util/RoleMethodAuthUtil.java [hunk 6]
```diff
@@ -46,14 +46,16 @@ public static boolean parseMethodSecurity(Method method, Principal principal, Pr
         } else { // try RolesAllowed
```
Comment (jim-krueger) [inline dev/com.ibm.ws.security.authorization.util/src/com/ibm/ws/security/authorization/util/RoleMethodAuthUtil.java:46]: Add 2022 to copyright above.

File: dev/com.ibm.ws.security.authorization.util/test/com/ibm/ws/security/authorization/util/RoleMethodAuthUtilTest.java [hunk 7]
```diff
@@ -63,7 +63,7 @@ public void nullPrincipal_RolesAllowedOnClass_noAnnotationOnMethod() throws Exce
         RoleMethodAuthUtil.parseMethodSecurity(method(ROLESALLOWED_ON_CLASS, "unannotated"),
```
Comment (jim-krueger) [inline dev/com.ibm.ws.security.authorization.util/test/com/ibm/ws/security/authorization/util/RoleMethodAuthUtilTest.java:63]: Add 2022 to copyright above.

File: dev/io.openliberty.restfulWS30.appSecurity/src/io/openliberty/restfulWS30/appSecurity/LibertyAuthFilter.java [hunk 8]
```diff
@@ -12,6 +12,11 @@
 
 import java.io.IOException;
 import java.lang.reflect.Method;
+
```
Comment (jim-krueger) [inline dev/io.openliberty.restfulWS30.appSecurity/src/io/openliberty/restfulWS30/appSecurity/LibertyAuthFilter.java:15]: Add 2022 to copyright above

**Comment by WhiteCat22** [comment]
#build - going to have restfulWS-3.0 errors

**Review by jim-krueger — APPROVED** [review]

**Comment by WhiteCat22** [comment]
#build #run-libby-bot

**Review by utle — APPROVED** [review]

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#build small change to `PrincipalBean` to ignore Unauthenticated subject in addition to `null` subject since I'm now setting the Unauthenticated Caller Subject

**Review by utle — APPROVED** [review]

**Review by jim-krueger — APPROVED** [review]

**Comment by teddyjtorres** [comment]
Except for the copyright year, the changes in dev/com.ibm.ws.security.mp.jwt.cdi/src/com/ibm/ws/security/mp/jwt/principal/PrincipalBean.java are good.

**Review by teddyjtorres — APPROVED** [review]
Except for the copyright year, the changes in dev/com.ibm.ws.security.mp.jwt.cdi/src/com/ibm/ws/security/mp/jwt/principal/PrincipalBean.java are good.

**Comment by WhiteCat22** [comment]
#run-libby-bot

**Pull request merged**